javascript/javascript1/week2/lord.js

```diff
@@ -0,0 +1,114 @@
+function getFullname(firstname, surname, useFormalName) {
+    if (!firstname || !surname) {
+        return "Please provide both firstname and surname.";
+    }
+    if (useFormalName) {
+        return `Lord ${firstname} ${surname}`;
+    } else {
+        return `${firstname} ${surname}`;
+    }
+}
+const fullname1 = getFullname("Benjamin", "Hughes", true); 
+const fullname2 = getFullname("Sophia", "Smith", false);  
+
+console.log(fullname1);
+console.log(fullname2);
+
+
+
+function getEventWeekday(daysFromToday) {
+    const weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+    const today = new Date().getDay();
+    const eventDay = (today + daysFromToday) % 7;
+    return weekdays[eventDay];
+}
+console.log(getEventWeekday(9)); 
+console.log(getEventWeekday(1));
+
+
+
+function whatToWear(temperature) {
+    if (temperature < 10) {
+        return "Wear a heavy jacket, scarf, and gloves.";
+    } else if (temperature < 20) {
+        return "Wear a light jacket and long pants.";
+    } else {
+        return "Shorts and a t-shirt are fine.";
+    }
+}
+console.log(whatToWear(18)); 
+
+
+
+
+const class07Students = [];
+function addStudentToClass(studentName) {
+    if (!studentName) {
+        console.log("You cannot add an empty name.");
+        return;
+    }
+    if (class07Students.includes(studentName)) {
+        console.log(`Student ${studentName} is already in the class.`);
+        return;
+    }
+    if (studentName === "Queen" || class07Students.length < 6) {
+        class07Students.push(studentName);
+    } else {
+        console.log("Cannot add more students to class 07.");
+    }
+}
+
+function getNumberOfStudents() {
+    return class07Students.length;
+}
+
+addStudentToClass("Benjamin");
+addStudentToClass("Sophia");
+addStudentToClass("Queen");
+addStudentToClass("Oliver");
+addStudentToClass("Ava");
+addStudentToClass("Emma");
+addStudentToClass("Liam"); 
+
+console.log(`Number of students: ${getNumberOfStudents()}`);

```

Comment:
It looks great all in all, and the early exits are great!
But the first condition is not 100% functional.
Therefore, I have a few comments:
`if (!studentName)` is correct, but it needs extra conditions, `if (!studentName || studentName.trim() === "")` where `.trim()` will return the whole string after deleting all the spaces.
` console.log("You cannot add an empty name.");
        return;` is not the best practice, and it is better to print outside of the function.
below you can find my suggestion.

```suggestion
const class07Students = [];
function addStudentToClass(studentName) {
  if (!studentName || studentName.trim() === "") {
    return "You cannot add an empty name.";
  }
  // the rest of the code.
}

addStudentToClass("");
addStudentToClass("   ");

console.log(addStudentToClass(" "));
```